cli/azd/pkg/azapi/standard_deployments.go

```diff
@@ -688,3 +691,129 @@ func convertFromStandardProvisioningState(state armresources.ProvisioningState)
 
 	return DeploymentProvisioningState("")
 }
+
+func (ds *StandardDeployments) ValidatePreflightToSubscription(
+	ctx context.Context,
+	subscriptionId string,
+	location string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+	options map[string]any,
+) error {
+	deploymentClient, err := ds.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	var rawResponse *http.Response
+	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
+
+	validate, err := deploymentClient.BeginValidateAtSubscriptionScope(
+		ctxWithResp, deploymentName,
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   armTemplate,
+				Parameters: parameters,
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+			Location: to.Ptr(location),
+			Tags:     tags,
+		}, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "subscription")
+	}
+
+	_, err = validate.PollUntilDone(ctx, nil)
+	if err != nil {
+		preflightError := createDeploymentError(err)
+		return fmt.Errorf(
+			"validating preflight to subscription:\n\nPreflight Error Details:\n%w",
+			preflightError,
+		)
+	}
+
+	return nil
+}
+
+type PreflightErrorResponse struct {
+	Error struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+		Details []struct {
+			Code    string `json:"code"`
+			Message string `json:"message"`
+		} `json:"details"`
+	} `json:"error"`
+}
```

Comment:
It looks like this basically captures the same ARM errors that we use in the `AzureDeploymentError` object that we already have.  Can we reuse that one instead?